llms.txt

```diff
@@ -11289,6 +11290,780 @@ Doc-Content: https://docs.kluster.ai/tutorials/klusterai-api/image-analysis.ipyn
+      "  \"choices\": [\n",
+      "    {\n",
+      "      \"finish_reason\": \"tool_calls\",\n",
+      "      \"index\": 0,\n",
+      "      \"logprobs\": null,\n",
+      "      \"message\": {\n",
+      "        \"content\": null,\n",
+      "        \"refusal\": null,\n",
+      "        \"role\": \"assistant\",\n",
+      "        \"audio\": null,\n",
+      "        \"function_call\": null,\n",
+      "        \"tool_calls\": [\n",
+      "          {\n",
+      "            \"id\": \"chatcmpl-tool-fa51f55519ff412bb1798262c3a5a797\",\n",
+      "            \"function\": {\n",
+      "              \"arguments\": \"{\\\"expression\\\": \\\"1337 * 42\\\"}\",\n",
+      "              \"name\": \"calculator\"\n",
+      "            },\n",
+      "            \"type\": \"function\"\n",
+      "          }\n",
+      "        ]\n",
+      "      },\n",
+      "      \"stop_reason\": 128008\n",
+      "    }\n",
+      "  ],\n",
+      "  \"created\": 1746741970,\n",
+      "  \"model\": \"klusterai/Meta-Llama-3.1-8B-Instruct-Turbo\",\n",
+      "  \"object\": \"chat.completion\",\n",
+      "  \"service_tier\": null,\n",
+      "  \"system_fingerprint\": null,\n",
+      "  \"usage\": {\n",
+      "    \"completion_tokens\": 20,\n",
+      "    \"prompt_tokens\": 252,\n",
+      "    \"total_tokens\": 272,\n",
+      "    \"completion_tokens_details\": null,\n",
+      "    \"prompt_tokens_details\": null\n",
+      "  },\n",
+      "  \"prompt_logprobs\": null\n",
+      "}\n"
+     ]
+    }
+   ],
+   "source": [
+    "def run_with_tools(prompt, tools, model=\"klusterai/Meta-Llama-3.1-8B-Instruct-Turbo\"):\n",
+    "    messages = [\n",
+    "        {\"role\": \"user\", \"content\": prompt}\n",
+    "    ]\n",
+    "    \n",
+    "    response = client.chat.completions.create(\n",
+    "        model=model,\n",
+    "        messages=messages,\n",
+    "        tools=tools,\n",
+    "        tool_choice=\"auto\"\n",
+    "    )\n",
+    "    \n",
+    "    return response\n",
+    "\n",
+    "# Define a calculator tool\n",
+    "calculator_tools = [\n",
+    "    {\n",
+    "        \"type\": \"function\",\n",
+    "        \"function\": {\n",
+    "            \"name\": \"calculator\",\n",
+    "            \"description\": \"Perform arithmetic calculations\",\n",
+    "            \"parameters\": {\n",
+    "                \"type\": \"object\",\n",
+    "                \"properties\": {\n",
+    "                    \"expression\": {\n",
+    "                        \"type\": \"string\",\n",
+    "                        \"description\": \"The mathematical expression to evaluate\"\n",
+    "                    }\n",
+    "                },\n",
+    "                \"required\": [\"expression\"]\n",
+    "            }\n",
+    "        }\n",
+    "    }\n",
+    "]\n",
+    "\n",
+    "# Test with a math problem\n",
+    "calculator_response = run_with_tools(\n",
+    "    \"What is 1337 multiplied by 42?\", \n",
```

Comment:
Ideally we  put this message near the top so folks can clearly see it and edit it when they run the notebook. Check how in other notebooks we put the prompt at the top etc